### PR TITLE
[ServiceBus] remove msrest

### PR DESCRIPTION
**Pull request by swathipil**
- Removing msrest + re-generating ServiceBusAdministrationClient
- Updated based on swagger at https://github.com/Azure/azure-rest-api-specs/blob/main/specification/servicebus/data-plane/Microsoft.ServiceBus/stable/2021-05/servicebus.json#L519
- `azure.core.exceptions.HttpResponseError`s are raised instead of `msrest.ValidationError`s.

- related to #26571

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 1]
```diff
@@ -6,10 +6,16 @@
 
 ### Breaking Changes
 
+- Operations on the sync and async `ServiceBusAdministrationClient` will now raise `azure.core.exceptions.HttpResponseError` everywhere `msrest.exceptions.ValidationErrors` were raised.
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:9]: While correct, I will not phrase it this way.
Client no longer do client side validation of parameters, and let server decide if a query is valid or not. As a consequence, you will no longer receive `msrest.exceptions.ValidationError` if the query is judged invalid by the client, but you may receive `azure.core.exceptions.HttpResponseError` if the server refuses the request. It's also avoiding the problem of over-validation when a client could refuse a request that the server would have accepted.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
     cast
 )
 from contextlib import contextmanager
-from msrest.serialization import TZ_UTC
+from azure.core.serialization import TZ_UTC
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py:27]: Python 3:
``` from datetime import timezone
TZ_UTC = timezone.utc
```
Or even you remove `TZ_UTC` and replace them all with `timezone.utc`

File: sdk/servicebus/azure-servicebus/azure/servicebus/management/_model_workaround.py
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/azure/servicebus/management/_model_workaround.py:1]: You should black those files

File: sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py [hunk 3]
```diff
@@ -68,7 +68,7 @@
 )
 from ._shared_key_policy import ServiceBusSharedKeyCredentialPolicy
 from ._generated._configuration import ServiceBusManagementClientConfiguration
-from ._generated._service_bus_management_client import (
+from ._generated._client import (
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py:71]:
```suggestion
from ._generated import (
```

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py [hunk 4]
```diff
@@ -583,7 +580,7 @@ async def get_topic_runtime_properties(
         if not entry.content:
             raise ResourceNotFoundError("Topic {} does not exist".format(topic_name))
         topic_description = TopicRuntimeProperties._from_internal_entity(
-            topic_name, entry.content.topic_description
+            topic_name, cast(TopicDescription, entry.content.topic_description)
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py:583]: Just curious, why does this have to be cast here does entry.content.topic_description have no type?

**Review by l0lawrence — COMMENTED** [review]
Looks good to me!

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py (hunk 4 above)
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py:583]: When it's returned, it's type `Optional[TopicDescription]`, but the _from_internal_entity method only takes type `TopicDescription` and not `None`, so had to cast.

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py [hunk 5]
```diff
@@ -524,7 +523,6 @@ def entry_to_qd(entry):
         get_next = functools.partial(
             get_next_template,
             functools.partial(self._impl.list_entities, constants.ENTITY_TYPE_QUEUES),
-            api_version=self._api_version,
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py:527]: passing in api version is no longer required ?

Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py:527]: it's passed into the client, then passed internally

**Review by kashifkhan — APPROVED** [review]

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 6]
```diff
@@ -4,12 +4,22 @@
 
 ### Features Added
 
+- The following methods have been added to `azure.servicebus.management.MessageCountDetails`: `deserialize`, `enable_additional_properties_sending`, `from_dict`, `is_xml_model`, `as_dict`, `serialize`.
```
Comment (lmazuel) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:7]: Don't document those in the ChangeLog please. While this is true, we don't want to advertised them so much.

**Review by lmazuel — APPROVED** [review]
Quick review LGTM!

**Pull request merged**